sycl/include/sycl/handler.hpp

```diff
@@ -918,8 +794,8 @@ class __SYCL_EXPORT handler {
         detail::KernelLambdaHasKernelHandlerArgT<KernelType,
                                                  LambdaArgType>::value;
 
-    KernelType *KernelPtr =
-        ResetHostKernel<KernelType, LambdaArgType, Dims>(KernelFunc);
+    MHostKernel.reset(
+        new detail::HostKernel<KernelType, LambdaArgType, Dims>(KernelFunc));
```

Comment:
should it be `= std::make_unique`?

Reply:
Originally I simply reverted this code to its older state, but applied `make_unique` in ed3589e3300cfaccbf16f56c9488a41113961d33